### PR TITLE
Feature/s2 index equals

### PR DESCRIPTION
**Pull request by hkernbach**
- added MultiPoint constructor (incl. tests & docu)
- added AQL GEO_EQUALS function (incl. test & docu)

### CODE REVIEW

File: arangod/Aql/Functions.cpp [hunk 1]
```diff
@@ -2995,6 +3030,70 @@ AqlValue Functions::GeoPoint(arangodb::aql::Query* query,
...
+AqlValue Functions::GeoMultiPoint(arangodb::aql::Query* query,
+                             transaction::Methods* trx,
+                             VPackFunctionParameters const& parameters) {
+
+  ValidateParameters(parameters, "GEO_MULTIPOINT", 1, 1);
+
+  size_t const n = parameters.size();
+
+  if (n < 1) {
+    // no parameters
```
Comment (jsteemann) [inline arangod/Aql/Functions.cpp]: do we need to issue a warning here too?

File: lib/Geo/ShapeContainer.cpp [hunk 2]
```diff
@@ -421,6 +421,125 @@ bool ShapeContainer::contains(ShapeContainer const* cc) const {
...
+  double lon2 = cc->longitude;
+
+  if (lat1 == lat2 && lon1 == lon2) {
+    return true;
+  }
+
+  return false;
+}
+
+bool ShapeContainer::equals(Coordinate const point, Coordinate const other) const {
```
Comment (jsteemann) [inline lib/Geo/ShapeContainer.cpp]: pass by reference?

File: arangod/Aql/Functions.cpp (hunk 1 above)
Comment (hkernbach) [reply arangod/Aql/Functions.cpp]: done

**Comment by Simran-B** [comment]
@hkernbach Are two polygons considered equal if they are congruent, but the points are in different order?

**Comment by hkernbach** [comment]
@Simran-B 
```
  // Return true if two polygons have exactly the same loops.  The loops must
  // appear in the same order, and corresponding loops must have the same
  // linear vertex ordering (i.e., cyclic rotations are not allowed).
```

**Comment by Simran-B** [comment]
@hkernbach Would be nice to have this in our documentation, although I'm not sure if explaining loops goes too far.

**Pull request merged**